.ci/pitest-survival-check-xml.groovy

```diff
@@ -107,7 +98,26 @@ private static int checkPitestReport(String profile, String flag) {
         final Node suppressedMutationNode = xmlParser.parse(suppressedMutationFileUri)
         suppressedMutations = getSuppressedMutations(suppressedMutationNode)
     }
-    return compareMutations(survivingMutations, suppressedMutations, flag)
+
+    final StringBuilder suppressionFileContent = new StringBuilder(1024)
+    suppressionFileContent.append('<?xml version="1.0" encoding="UTF-8"?>\n<suppressedMutations>')
+
+    survivingMutations.each {
+        suppressionFileContent.append(it.toXmlString())
+    }
+    if (survivingMutations.isEmpty()) {
+        suppressionFileContent.append('\n')
+    }
+    suppressionFileContent.append('</suppressedMutations>\n')
+
+    if (suppressionFile.exists()) {
+        suppressionFile.write(suppressionFileContent.toString())
+    } else if (!survivingMutations.isEmpty()) {
```

Comment:
It seems to me this condition should wrap everything as it is used the only thing for the loops. If `survivingMutations` is empty, it is pointless to populate the file content with anything.

Also, if file goes from having something to having nothing, it won't delete the file. If we are going to automatically update things, it should erase the file if it is completely empty with no surviving mutations left.

Reply:
Done, modified logic to cover file deletion.